detox/ios/DetoxXCUITestRunner/DetoxXCUITestRunner/Handlers/ActionHandler.swift

```diff
@@ -7,48 +7,96 @@ import Foundation
 import XCTest
 
 class ActionHandler {
-  func handle(from params: InvocationParams, on element: XCUIElement) throws {
-
-    let exists = element.waitForExistence(timeout: .defaultTimeout)
-    DTXAssert(
-      exists,
-      "Action failed, element with matcher `\(params.matcherDescription)` does not exist"
-    )
+    
+    func findElement(from params: InvocationParams, predicateHandler: PredicateHandler) -> XCUIElement {
+        
+        let element = predicateHandler.findElement(using: params)
 
+      let exists = element.waitForExistence(timeout: .defaultTimeout)
+      DTXAssert(
+        exists,
+        "Action failed, element with matcher `\(params.matcherDescription)` does not exist"
+      )
+        return element;
+    }
+    
+    func getNormalizedCoordinate(from params: InvocationParams) throws -> XCUICoordinate {
+        do {
+            guard let x = Int(params.params?.first ?? "100"), let y = Int(params.params?[1] ?? "100") else {
+                throw Error.missingTypeTextParam
```

Comment:
you're throwing this error but catching it and hiding it (with `failedToTapDeviceByCoordinates`, see below).

Reply:
also this is not the right error to throw, you might want to make this error more generic by introducing params to it, e.g. `missingParam(paramName, paramType)`

Reply:
I removed the error because we want to allow this case, I just need the fallback